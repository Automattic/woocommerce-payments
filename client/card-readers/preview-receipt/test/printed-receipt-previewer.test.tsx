/**
 * External dependencies
 */
import { render } from '@testing-library/react';
import React from 'react';

/**
 * Internal dependencies
 */
import PrintedReceiptPreviewer from '../previewer';

describe( 'PrintedReceiptPreviewer', () => {
	it( 'should render correctly', () => {
		const expected = 'test';
		const { container, getByTitle } = render(
			<PrintedReceiptPreviewer receiptHtml={ expected } />
		);
		const iframeElement = getByTitle(
			'Preview Receipt'
		) as HTMLIFrameElement;

		expect( iframeElement?.contentWindow?.document.body.innerHTML ).toEqual(
			`<div>${ expected }</div>`
		);
		expect( container ).toMatchSnapshot();
	} );
} );
