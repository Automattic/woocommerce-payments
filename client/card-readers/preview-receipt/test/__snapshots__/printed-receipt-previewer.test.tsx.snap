// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrintedReceiptPreviewer should render correctly 1`] = `
<div>
  <iframe
    class="card-readers-preview-receipt__preview"
    title="Preview Receipt"
  />
</div>
`;
